test/unit/features/options/directives.spec.js

```diff
@@ -287,4 +287,28 @@ describe('Options directives', () => {
     }).$mount()
     vm.key = 'bar'
   })
+
+  it('deep object as dynamic arguments', done => {
+    const vm = new Vue({
+      template: `<div v-my:[deep.key]="1"/>`,
+      data: {
+        deep: {
+          key: 'foo'
+        }
+      },
+      directives: {
+        my: {
+          bind(el, binding) {
+            expect(binding.arg).toBe('foo')
+          },
+          update(el, binding) {
+            expect(binding.arg).toBe('bar')
+            expect(binding.oldArg).toBe('foo')
+            done()
+          }
+        }
+      }
+    }).$mount()
+    vm.deep.key = 'bar'
+  })
```

Comment:
We may need more test case covering more dynamic arg expression. And maybe we shouldn't allow `]` inside the expression in the first place because that would make parsing a lot more complicated.

Thoughts? @yyx990803

Reply:
I added a test case with dynamic arguments like `deep.a.b`.